pkg/distributor/distributor_test.go

```diff
@@ -2208,6 +2211,13 @@ func TestDistributor_MetricsForLabelMatchers(t *testing.T) {
 			},
 			expectedIngesters: 3,
 		},
+		"should error out if max series per query is reached": {
+			matchers: []*labels.Matcher{
+				mustNewMatcher(labels.MatchEqual, model.MetricNameLabel, "test_1"),
+			},
+			maxSeriesPerQuery: 1,
+			expectedError:     validation.NewLimitError("the query exceeded the maximum number of series (limit: 1 series) (err-mimir-max-series-per-query). Consider reducing the time range and/or number of series selected by the query. One way to reduce the number of selected series is to add more label matchers to the query. Otherwise, to adjust the related per-tenant limit, configure -querier.max-fetched-series-per-query, or contact your service administrator."),
```

Comment:
nit: I don't think the message needs to match exactly with `require.ErrorIs`

Reply:
gives me an error if I reduce it : 
```
Error:      	Target error should be in err chain:
            	            	expected: "the query exceeded the maximum number of series (limit: 1 series) (err-mimir-max-series-per-query)."
            	            	in chain: "the query exceeded the maximum number of series (limit: 1 series) (err-mimir-max-series-per-query). Consider reducing the time range and/or number of series selected by the query. One way to reduce the number of selected series is to add more label matchers to the query. Otherwise, to adjust the related per-tenant limit, configure -querier.max-fetched-series-per-query, or contact your service administrator."
```